.github/workflows/main.yml

```diff
@@ -55,13 +55,17 @@ jobs:
 
   test:
     runs-on: ubuntu-latest
+    strategy: 
+      matrix:
+        python: [py32, py39, py310, py311] 
+        pytorch: [torch20, torch21, torch22, torch23, torch24]
     steps:
       - uses: actions/checkout@v4
       - uses: prefix-dev/setup-pixi@v0.8.1
         with:
           pixi-version: v0.25.0
-          cache: true
+          cache: false
           frozen: true
-          environments: default
-          activate-environment: default
+          environments: ${{ matrix.python }}-${{ matrix.pytorch }}
```

Comment:
Can we build this environment dynamically? E.g. I don't want to add 4 x 5 environments to the pixi.toml

Reply:
I couldn't find anything about it in pixi's docs just yet, I also would rather do it dynamically.

Reply:
Maybe we can edit `pixi.toml` during runtime (after checkout, before `setup-pixi`) and "inject" the environment into the file then. It will be temporary, as we don't run a commit action in this workflow.